src/snowflake/snowpark/mock/plan.py

```diff
@@ -808,11 +818,275 @@ def outer_join(base_df):
             frac=source_plan.probability_fraction,
             random_state=source_plan.seed,
         )
-    elif isinstance(source_plan, CreateViewCommand):
+    if isinstance(source_plan, CreateViewCommand):
         from_df = execute_mock_plan(source_plan.child, expr_to_alias)
         view_name = source_plan.name
         entity_registry.create_or_replace_view(source_plan.child, view_name)
         return from_df
+
+    if isinstance(source_plan, TableUpdate):
+        target = entity_registry.read_table(source_plan.table_name)
+        ROW_ID = "row_id_" + generate_random_alphanumeric()
+        target.insert(0, ROW_ID, range(len(target)))
+
+        multi_joins = 0
```

Comment:
this line feels unnecessary as it will always be overwritten by line 863: `multi_joins = matched_count.where(lambda x: x > 1).count()`